tests/test_fact_manager.py

```diff
@@ -0,0 +1,439 @@
+        await fm.add(test_fact)
+
+
+@pytest.mark.asyncio
+async def test_fm_add_psycopg2_errors(test_fm_fx):
+    """
+    Verify an error is raised if a fact already exists.
+    """
+    fm = test_fm_fx
+    test_fact = Fact('test', 'en', 'This is a test fact for pytest',
+                     None, 'Shatt', None, 'Shatt', False)
+
+    # Intentionally add a fact with the same name:
+    with pytest.raises(psycopg2.IntegrityError):
+        await fm.add(test_fact)
+
+
+@pytest.mark.asyncio
+async def test_fm_delete_mfd_check(test_fm_fx):
+    """
+    Verify deleting a fact marked mfd succeeds
+    """
+    fm = test_fm_fx
+
+    test_fact = Fact('test127', 'en', 'This is a test fact for pytest',
+                     None, 'Shatt', None, 'Shatt', False)
+
+    await fm.add(test_fact)
+    assert await fm.exists('test127', 'en')
+
+    # Mark it for deletion
+    await fm.mfd('test127', 'en')
+
+    # Attempt to delete it:
+    await fm.delete('test127', 'en')
+
+    # ..and verify it is deleted.
+    assert not await fm.exists('test127', 'en')
+
+
+@pytest.mark.asyncio
+async def test_fm_delete_no_mfd_flag(test_fm_fx):
+    """
+    Verify deleting a fact NOT marked mfd fails, throwing a psycopg2.ProgrammingError
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(psycopg2.ProgrammingError):
+        await fm.delete('prep', 'en')
+
+
+@pytest.mark.asyncio
+async def test_mfdlist_returns_list(test_fm_fx):
+    """
+    mfd_list method must return a list.
+    """
+    fm = test_fm_fx
+
+    assert isinstance(await fm.mfd_list(), list)
+
+
+@pytest.mark.asyncio
+async def test_fm_destroy(test_fm_fx):
+    """
+    Verify _destroy function works properly.
+    """
+    fm = test_fm_fx
+
+    await fm._destroy('test', 'ru')
+
+
+@pytest.mark.asyncio
+async def test_edit_message_bad_fact(test_fm_fx):
+    """
+    Verify attempt to edit a fact that does not exist throws exception.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('notafact', 'nope', 'Shatt', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_edit_message(test_fm_fx):
+    """
+    Test edit_message functionally works.
+    """
+    fm = test_fm_fx
+    await fm.edit_message('test', 'en', 'Shatt', 'This fact has been edited during testing.')
+
+    result = await fm.find('test', 'en')
+
+    assert result.message == 'This fact has been edited during testing.'
+
+
+@pytest.mark.asyncio
+async def test_edit_message_failure(test_fm_fx):
+    """
+    Verify an error is raised if an edit on a non-existent fact is attempted.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('nopeavi', 'nope', 'Kovacs', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_fact_exists(test_fm_fx):
+    """
+    Verify a fact looked up with the exists method returns true.
+    """
+    fm = test_fm_fx
+    assert bool(await fm.exists('stats', 'en'))
```

Comment:
Casting the result to a `bool` seems to defeat the test's purpose of ensuring the method returns a `bool`. Almost any value aside from `None` or `False` will cast to `True`.

Reply:
further, boolean casts are redundant as python will already cast them for use in the assertion statement.

Reply:
fixed in 83935a7